src/V3Hasher.cpp

```diff
@@ -33,9 +33,29 @@ class HasherVisitor final : public VNVisitorConst {
     // STATE
     V3Hash m_hash;  // Hash value accumulator
     const bool m_cacheInUser4;  // Use user4 to cache each V3Hash?
+    std::vector<AstNode*> m_stack;  // Keeps track of some visited nodes to prevent
```

Comment:
Perhaps instead just a std::set on the cfunc?

Simpler and avoids O(n^2) on stack depth.

Reply:
I'd expect you need a few dozen elements for a set to be faster. So I'm not convinced which one wins on average. Though I admit, I didn't measure.
But I was also thinking about situations like:
```
class C;
    task f;
        ...
    endtask
    task g;
        f();
    endtask
endclass
```
With a set, after the first `f`, any subsequent occurences of it would get hashed to `VHash{1}`. Not sure if it's a problem, but it may be safer to do a map-based cache.